File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/data/entity/Team.java
package com.benoitquenaudon.tvfoot.red.app.data.entity;

import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import javax.annotation.Nullable;

@AutoValue public abstract class Team {
  public static final String DEFAULT_CODE = "default";

  public static TypeAdapter<Team> typeAdapter(Gson gson) {
    return new AutoValue_Team.GsonTypeAdapter(gson);
  }

  @Nullable public abstract String code();

  @Nullable public abstract String name();

  @Nullable public abstract String fullname();

  @Nullable public abstract String city();

  @Nullable public abstract String country();

  @Nullable public abstract String url();

  @Nullable public abstract String stadium();

  @Nullable public abstract String twitter();

  @Nullable public abstract String type();

  public boolean isEmpty() {
    return name() == null;
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/data/entity/Team.kt
package com.benoitquenaudon.tvfoot.red.app.data.entity

import com.benoitquenaudon.tvfoot.red.util.stripAccents

data class Team(
    val id: String?,
    val code: String?,
    val name: String?,
    val fullname: String?,
    val city: String?,
    val country: String?,
    val url: String?,
    val stadium: String?,
    val twitter: String?,
    val type: String?
) {
  val isEmpty: Boolean
    get() = name == null

  companion object Constant {
    val DEFAULT_CODE = "default"
  }

  val logoPath: String
    get() = if (code == null || type == null) {
      "/images/teams/default/large/default.png"
    } else {
      when (type) {
        "nation" -> String.format("/images/teams/nations/large/%s.png", code.toLowerCase())
        "club" -> {
          val country = checkNotNull(country) { "team's country should not be null" }
          "/images/teams/${country.stripAccents()}/large/${code.toLowerCase()}.png"
        }
        else -> throw IllegalStateException("Unknown type " + type)
      }
    }
}
